crates/oxc_minifier/src/ast_passes/peephole_substitute_alternate_syntax.rs

```diff
@@ -265,6 +276,123 @@ impl<'a, 'b> PeepholeSubstituteAlternateSyntax {
         self.changed = true;
     }
 
+    /// Compress `foo === null || foo === undefined` into `foo == null`.
+    ///
+    /// `foo === null || foo === undefined` => `foo == null`
+    /// `foo !== null && foo !== undefined` => `foo != null`
+    ///
+    /// This compression assumes that `document.all` is a normal object.
+    /// If that assumption does not hold, this compression is not allowed.
+    /// - `document.all === null || document.all === undefined` is `false`
+    /// - `document.all == null` is `true`
```

Comment:
terser makes this assumption.
https://github.com/terser/terser#:~:text=document.all%20is%20not%20%3D%3D%20null